<?xml version="1.0" encoding="UTF-8" ?>
<ui version="4.0">
	<class>ImageSegmentationClass</class>
	<widget class="QMainWindow" name="ImageSegmentationClass">
		<property name="geometry">
			<rect>
				<x>0</x>
				<y>0</y>
				<width>1024</width>
				<height>591</height>
			</rect>
		</property>
		<property name="sizePolicy">
			<sizepolicy hsizetype="Expanding" vsizetype="Expanding">
				<horstretch>0</horstretch>
				<verstretch>0</verstretch>
			</sizepolicy>
		</property>
		<property name="windowTitle">
			<string>ImageSegmentation</string>
		</property>
		<widget class="QWidget" name="centralWidget">
			<property name="sizePolicy">
				<sizepolicy hsizetype="Expanding" vsizetype="Expanding">
					<horstretch>0</horstretch>
					<verstretch>0</verstretch>
				</sizepolicy>
			</property>
			<layout class="QGridLayout" name="gridLayout_5">
				<item row="0" column="0">
					<widget class="QScrollArea" name="scrollArea">
						<property name="sizePolicy">
							<sizepolicy hsizetype="Expanding" vsizetype="Preferred">
								<horstretch>0</horstretch>
								<verstretch>0</verstretch>
							</sizepolicy>
						</property>
						<property name="widgetResizable">
							<bool>true</bool>
						</property>
						<widget class="QWidget" name="scrollAreaWidgetContents">
							<property name="geometry">
								<rect>
									<x>0</x>
									<y>0</y>
									<width>643</width>
									<height>498</height>
								</rect>
							</property>
							<layout class="QGridLayout" name="gridLayout_2">
								<item row="0" column="0">
									<widget class="QLabel" name="imageLabel">
										<property name="sizePolicy">
											<sizepolicy hsizetype="Maximum" vsizetype="Maximum">
												<horstretch>0</horstretch>
												<verstretch>0</verstretch>
											</sizepolicy>
										</property>
										<property name="text">
											<string>TextLabel</string>
										</property>
									</widget>
								</item>
							</layout>
						</widget>
					</widget>
				</item>
				<item row="0" column="1">
					<widget class="QTabWidget" name="tabWidget">
						<property name="sizePolicy">
							<sizepolicy hsizetype="Preferred" vsizetype="Preferred">
								<horstretch>0</horstretch>
								<verstretch>0</verstretch>
							</sizepolicy>
						</property>
						<property name="maximumSize">
							<size>
								<width>350</width>
								<height>500</height>
							</size>
						</property>
						<property name="currentIndex">
							<number>0</number>
						</property>
						<widget class="QWidget" name="tab">
							<attribute name="title">
								<string>Tools</string>
							</attribute>
							<layout class="QGridLayout" name="gridLayout_4">
								<item row="0" column="0">
									<widget class="QGroupBox" name="groupBox_2">
										<property name="maximumSize">
											<size>
												<width>16777215</width>
												<height>200</height>
											</size>
										</property>
										<property name="title">
											<string>Parameters</string>
										</property>
										<layout class="QVBoxLayout" name="verticalLayout_2">
											<item>
												<layout class="QHBoxLayout" name="horizontalLayout_3">
													<item>
														<widget class="QLabel" name="label_2">
															<property name="text">
																<string>Scale</string>
															</property>
														</widget>
													</item>
													<item>
														<widget class="QDoubleSpinBox" name="doubleSpinBoxScale" />
													</item>
													<item>
														<widget class="QPushButton" name="pushButtonScale">
															<property name="text">
																<string>Scale</string>
															</property>
														</widget>
													</item>
												</layout>
											</item>
											<item>
												<layout class="QHBoxLayout" name="horizontalLayout">
													<item>
														<widget class="QLabel" name="label">
															<property name="text">
																<string>Î»</string>
															</property>
														</widget>
													</item>
													<item>
														<widget class="QDoubleSpinBox" name="doubleSpinBoxLambda">
															<property name="maximum">
																<double>100000000.000000000000000</double>
															</property>
														</widget>
													</item>
												</layout>
											</item>
										</layout>
									</widget>
								</item>
								<item row="1" column="0">
									<layout class="QHBoxLayout" name="horizontalLayout_17">
										<item>
											<widget class="QGroupBox" name="groupBox_6">
												<property name="sizePolicy">
													<sizepolicy hsizetype="Preferred" vsizetype="Preferred">
														<horstretch>0</horstretch>
														<verstretch>0</verstretch>
													</sizepolicy>
												</property>
												<property name="maximumSize">
													<size>
														<width>16777215</width>
														<height>150</height>
													</size>
												</property>
												<property name="title">
													<string>Object type</string>
												</property>
												<layout class="QGridLayout" name="gridLayout_6">
													<item row="0" column="0">
														<layout class="QVBoxLayout" name="verticalLayout_13">
															<item>
																<widget class="QToolButton" name="toolButtonDark">
																	<property name="text">
																		<string>Dark</string>
																	</property>
																	<property name="checkable">
																		<bool>true</bool>
																	</property>
																</widget>
															</item>
															<item>
																<widget class="QToolButton" name="toolButtonLight">
																	<property name="text">
																		<string>Light</string>
																	</property>
																	<property name="checkable">
																		<bool>true</bool>
																	</property>
																</widget>
															</item>
															<item>
																<widget class="QToolButton" name="toolButtonAuto">
																	<property name="text">
																		<string>Auto</string>
																	</property>
																	<property name="checkable">
																		<bool>true</bool>
																	</property>
																</widget>
															</item>
														</layout>
													</item>
												</layout>
											</widget>
										</item>
										<item>
											<widget class="QGroupBox" name="groupBox_4">
												<property name="sizePolicy">
													<sizepolicy hsizetype="Preferred" vsizetype="Preferred">
														<horstretch>0</horstretch>
														<verstretch>0</verstretch>
													</sizepolicy>
												</property>
												<property name="maximumSize">
													<size>
														<width>16777215</width>
														<height>150</height>
													</size>
												</property>
												<property name="title">
													<string>Seed parameters</string>
												</property>
												<layout class="QHBoxLayout" name="horizontalLayout_8">
													<item>
														<layout class="QVBoxLayout" name="verticalLayout_9">
															<item>
																<layout class="QHBoxLayout" name="horizontalLayout_12">
																	<item>
																		<widget class="QLabel" name="label_8">
																			<property name="sizePolicy">
																				<sizepolicy hsizetype="Minimum" vsizetype="Minimum">
																					<horstretch>0</horstretch>
																					<verstretch>0</verstretch>
																				</sizepolicy>
																			</property>
																			<property name="text">
																				<string>Noise threshold</string>
																			</property>
																		</widget>
																	</item>
																	<item>
																		<widget class="QSpinBox" name="spinBoxThreshold">
																			<property name="maximum">
																				<number>1000000</number>
																			</property>
																			<property name="value">
																				<number>100</number>
																			</property>
																		</widget>
																	</item>
																</layout>
															</item>
															<item>
																<layout class="QHBoxLayout" name="horizontalLayout_9">
																	<item>
																		<widget class="QLabel" name="label_5">
																			<property name="sizePolicy">
																				<sizepolicy hsizetype="Minimum" vsizetype="Minimum">
																					<horstretch>0</horstretch>
																					<verstretch>0</verstretch>
																				</sizepolicy>
																			</property>
																			<property name="text">
																				<string>Background seed</string>
																			</property>
																		</widget>
																	</item>
																	<item>
																		<widget class="QDoubleSpinBox" name="doubleSpinBoxBackground">
																			<property name="maximum">
																				<double>256.000000000000000</double>
																			</property>
																		</widget>
																	</item>
																</layout>
															</item>
															<item>
																<layout class="QHBoxLayout" name="horizontalLayout_11">
																	<item>
																		<widget class="QLabel" name="label_7">
																			<property name="sizePolicy">
																				<sizepolicy hsizetype="Minimum" vsizetype="Minimum">
																					<horstretch>0</horstretch>
																					<verstretch>0</verstretch>
																				</sizepolicy>
																			</property>
																			<property name="text">
																				<string>Object seed</string>
																			</property>
																		</widget>
																	</item>
																	<item>
																		<widget class="QDoubleSpinBox" name="doubleSpinBoxObject">
																			<property name="maximum">
																				<double>256.000000000000000</double>
																			</property>
																		</widget>
																	</item>
																</layout>
															</item>
														</layout>
													</item>
												</layout>
											</widget>
										</item>
									</layout>
								</item>
								<item row="2" column="0">
									<widget class="QGroupBox" name="groupBox_3">
										<property name="maximumSize">
											<size>
												<width>16777215</width>
												<height>350</height>
											</size>
										</property>
										<property name="title">
											<string>Area</string>
										</property>
										<layout class="QVBoxLayout" name="verticalLayout">
											<item>
												<layout class="QHBoxLayout" name="horizontalLayout_4">
													<item>
														<widget class="QLabel" name="label_3">
															<property name="text">
																<string>Image</string>
															</property>
														</widget>
													</item>
													<item>
														<widget class="QSpinBox" name="spinBoxImageArea">
															<property name="sizePolicy">
																<sizepolicy hsizetype="Maximum" vsizetype="Fixed">
																	<horstretch>0</horstretch>
																	<verstretch>0</verstretch>
																</sizepolicy>
															</property>
															<property name="readOnly">
																<bool>true</bool>
															</property>
															<property name="buttonSymbols">
																<enum>QAbstractSpinBox::ButtonSymbols::NoButtons</enum>
															</property>
															<property name="maximum">
																<number>10000000</number>
															</property>
														</widget>
													</item>
												</layout>
											</item>
											<item>
												<layout class="QHBoxLayout" name="horizontalLayout_6">
													<item>
														<widget class="QLabel" name="label_4">
															<property name="text">
																<string>Objects</string>
															</property>
														</widget>
													</item>
													<item>
														<spacer name="horizontalSpacer_4">
															<property name="orientation">
																<enum>Qt::Orientation::Horizontal</enum>
															</property>
															<property name="sizeHint" stdset="0">
																<size>
																	<width>40</width>
																	<height>20</height>
																</size>
															</property>
														</spacer>
													</item>
													<item>
														<widget class="QSpinBox" name="spinBoxObjectArea">
															<property name="sizePolicy">
																<sizepolicy hsizetype="Maximum" vsizetype="Fixed">
																	<horstretch>0</horstretch>
																	<verstretch>0</verstretch>
																</sizepolicy>
															</property>
															<property name="readOnly">
																<bool>true</bool>
															</property>
															<property name="buttonSymbols">
																<enum>QAbstractSpinBox::ButtonSymbols::NoButtons</enum>
															</property>
															<property name="maximum">
																<number>10000000</number>
															</property>
														</widget>
													</item>
													<item>
														<widget class="QLabel" name="labelObjectArea">
															<property name="sizePolicy">
																<sizepolicy hsizetype="Minimum" vsizetype="Preferred">
																	<horstretch>0</horstretch>
																	<verstretch>0</verstretch>
																</sizepolicy>
															</property>
															<property name="text">
																<string>0%</string>
															</property>
														</widget>
													</item>
												</layout>
											</item>
											<item>
												<widget class="QLabel" name="labelFeretResults">
													<property name="text">
														<string>Feret</string>
													</property>
												</widget>
											</item>
											<item>
												<layout class="QHBoxLayout" name="horizontalLayout_7">
													<item>
														<widget class="QGroupBox" name="groupBox">
															<property name="sizePolicy">
																<sizepolicy hsizetype="Expanding" vsizetype="Preferred">
																	<horstretch>0</horstretch>
																	<verstretch>0</verstretch>
																</sizepolicy>
															</property>
															<property name="title">
																<string>Show</string>
															</property>
															<layout class="QGridLayout" name="gridLayout">
																<item row="0" column="0">
																	<layout class="QVBoxLayout" name="verticalLayout_4">
																		<item>
																			<layout class="QHBoxLayout" name="horizontalLayout_5">
																				<item>
																					<widget class="QToolButton" name="toolButtonFeret">
																						<property name="text">
																							<string>Feret</string>
																						</property>
																						<property name="checkable">
																							<bool>true</bool>
																						</property>
																					</widget>
																				</item>
																				<item>
																					<widget class="QToolButton" name="toolButtonEllipse">
																						<property name="text">
																							<string>Ellipse</string>
																						</property>
																						<property name="checkable">
																							<bool>true</bool>
																						</property>
																					</widget>
																				</item>
																				<item>
																					<widget class="QToolButton" name="toolButtonMBR">
																						<property name="text">
																							<string>MBR</string>
																						</property>
																						<property name="checkable">
																							<bool>true</bool>
																						</property>
																					</widget>
																				</item>
																			</layout>
																		</item>
																		<item>
																			<layout class="QHBoxLayout" name="horizontalLayout_2">
																				<item>
																					<widget class="QToolButton" name="toolButtonEdge">
																						<property name="text">
																							<string>Edge</string>
																						</property>
																						<property name="checkable">
																							<bool>true</bool>
																						</property>
																					</widget>
																				</item>
																				<item>
																					<widget class="QToolButton" name="toolButtonObject">
																						<property name="text">
																							<string>Object</string>
																						</property>
																						<property name="checkable">
																							<bool>true</bool>
																						</property>
																					</widget>
																				</item>
																			</layout>
																		</item>
																	</layout>
																</item>
															</layout>
														</widget>
													</item>
													<item>
														<widget class="QGroupBox" name="groupBox_7">
															<property name="title">
																<string>Select ROI</string>
															</property>
															<layout class="QHBoxLayout" name="horizontalLayout_10">
																<item>
																	<widget class="QToolButton" name="toolButtonSelectROIRectangle">
																		<property name="text">
																			<string>Rectangle</string>
																		</property>
																		<property name="checkable">
																			<bool>true</bool>
																		</property>
																	</widget>
																</item>
																<item>
																	<widget class="QToolButton" name="toolButtonSelectROICustom">
																		<property name="enabled">
																			<bool>false</bool>
																		</property>
																		<property name="text">
																			<string>Custom</string>
																		</property>
																		<property name="checkable">
																			<bool>true</bool>
																		</property>
																	</widget>
																</item>
															</layout>
														</widget>
													</item>
												</layout>
											</item>
											<item>
												<widget class="QPushButton" name="pushButtonProcess">
													<property name="text">
														<string>Process</string>
													</property>
												</widget>
											</item>
										</layout>
									</widget>
								</item>
							</layout>
						</widget>
						<widget class="QWidget" name="tab_2">
							<attribute name="title">
								<string>Output info</string>
							</attribute>
							<layout class="QVBoxLayout" name="verticalLayout_10">
								<item>
									<layout class="QHBoxLayout" name="horizontalLayout_13">
										<item>
											<widget class="QLabel" name="label_9">
												<property name="text">
													<string>Pixel size (nm)</string>
												</property>
											</widget>
										</item>
										<item>
											<widget class="QDoubleSpinBox" name="doubleSpinBoxPixelSize">
												<property name="maximum">
													<double>100000000.000000000000000</double>
												</property>
											</widget>
										</item>
									</layout>
								</item>
								<item>
									<widget class="QGroupBox" name="groupBox_5">
										<property name="title">
											<string>Object info</string>
										</property>
										<layout class="QGridLayout" name="gridLayout_3">
											<item row="0" column="1">
												<layout class="QVBoxLayout" name="verticalLayout_6">
													<item>
														<widget class="QCheckBox" name="checkBoxPerimeter">
															<property name="text">
																<string>Perimeter </string>
															</property>
														</widget>
													</item>
													<item>
														<widget class="QCheckBox" name="checkBoxFeret">
															<property name="text">
																<string>Feret's Diameter</string>
															</property>
														</widget>
													</item>
													<item>
														<widget class="QCheckBox" name="checkBoxEllipseRatio">
															<property name="text">
																<string>Ellipse Ratio</string>
															</property>
														</widget>
													</item>
													<item>
														<widget class="QCheckBox" name="checkBoxLWRatio">
															<property name="text">
																<string>L/W Ratio</string>
															</property>
														</widget>
													</item>
													<item>
														<widget class="QCheckBox" name="checkBoxAspectRatio">
															<property name="text">
																<string>Aspect Ratio</string>
															</property>
														</widget>
													</item>
												</layout>
											</item>
											<item row="0" column="0">
												<layout class="QVBoxLayout" name="verticalLayout_7">
													<item>
														<widget class="QCheckBox" name="checkBoxArea">
															<property name="text">
																<string>Area</string>
															</property>
														</widget>
													</item>
													<item>
														<widget class="QCheckBox" name="checkBoxMeanGrayValue">
															<property name="text">
																<string>Mean Gray Value</string>
															</property>
														</widget>
													</item>
													<item>
														<widget class="QCheckBox" name="checkBoxModalGrayValue">
															<property name="text">
																<string>Modal Gray Value</string>
															</property>
														</widget>
													</item>
													<item>
														<widget class="QCheckBox" name="checkBoxMinMax">
															<property name="text">
																<string>Min &amp; Max Gray Level</string>
															</property>
														</widget>
													</item>
													<item>
														<widget class="QCheckBox" name="checkBoxCentroid">
															<property name="text">
																<string>Centroid</string>
															</property>
														</widget>
													</item>
													<item>
														<widget class="QCheckBox" name="checkBoxCircularity">
															<property name="text">
																<string>Circularity</string>
															</property>
														</widget>
													</item>
												</layout>
											</item>
											<item row="1" column="0">
												<spacer name="verticalSpacer">
													<property name="orientation">
														<enum>Qt::Orientation::Vertical</enum>
													</property>
													<property name="sizeHint" stdset="0">
														<size>
															<width>20</width>
															<height>40</height>
														</size>
													</property>
												</spacer>
											</item>
										</layout>
									</widget>
								</item>
							</layout>
						</widget>
					</widget>
				</item>
			</layout>
		</widget>
		<widget class="QMenuBar" name="menuBar">
			<property name="geometry">
				<rect>
					<x>0</x>
					<y>0</y>
					<width>1024</width>
					<height>25</height>
				</rect>
			</property>
			<widget class="QMenu" name="menuFile">
				<property name="title">
					<string>File</string>
				</property>
				<addaction name="actionOpen" />
				<addaction name="actionSave" />
				<addaction name="actionProcessFolder" />
				<addaction name="actionSaveObject" />
				<addaction name="actionSaveAllStates" />
				<addaction name="actionSaveInfo" />
			</widget>
			<addaction name="menuFile" />
		</widget>
		<widget class="QToolBar" name="mainToolBar">
			<attribute name="toolBarArea">
				<enum>TopToolBarArea</enum>
			</attribute>
			<attribute name="toolBarBreak">
				<bool>false</bool>
			</attribute>
		</widget>
		<widget class="QStatusBar" name="statusBar" />
		<action name="actionOpen">
			<property name="text">
				<string>Open</string>
			</property>
		</action>
		<action name="actionSave">
			<property name="text">
				<string>Save</string>
			</property>
		</action>
		<action name="actionProcessFolder">
			<property name="text">
				<string>Process Folder</string>
			</property>
		</action>
		<action name="actionSaveObject">
			<property name="text">
				<string>Save object...</string>
			</property>
		</action>
		<action name="actionSaveInfo">
			<property name="text">
				<string>Save info...</string>
			</property>
		</action>
		<action name="actionSaveAllStates">
			<property name="text">
				<string>Save all states...</string>
			</property>
		</action>
	</widget>
	<layoutdefault spacing="6" margin="11" />
	<resources>
		<include location="ImageSegmentation.qrc" />
	</resources>
	<connections />
</ui>
